SPECS/dnf5/dnf5.spec

```diff
@@ -0,0 +1,649 @@
+%global project_version_major 5
+%global project_version_minor 0
+%global project_version_patch 14
+# ========== versions of dependencies ==========
+%global libmodulemd_version 2.5.0
+%global librepo_version 1.15.0
+%global libsolv_version 0.7.21
+%global sqlite_version 3.35.0
+%global swig_version 4
+%global zchunk_version 0.9.11
+# ========== build options ==========
+%bcond_without dnf5daemon_client
+%bcond_without dnf5daemon_server
+%bcond_without libdnf_cli
+%bcond_without dnf5
+%bcond_without dnf5_plugins
+%bcond_without plugin_actions
+%bcond_without python_plugins_loader
+%bcond_without comps
+%bcond_without modulemd
+%bcond_without zchunk
+%bcond_with    html
+%bcond_with    man
+# TODO Go bindings fail to build, disable for now
+%bcond_with    go
+%bcond_with    perl5
+%bcond_without python3
+%bcond_with    ruby
+%bcond_with    clang
+%bcond_with    sanitizers
+%bcond_without tests
+%bcond_with    performance_tests
+%bcond_with    dnf5daemon_tests
+%if %{with clang}
+    %global toolchain clang
+%endif
+Summary:        Command-line package manager
+Name:           dnf5
+Version:        %{project_version_major}.%{project_version_minor}.%{project_version_patch}
+Release:        1%{?dist}
+License:        GPLv2+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/rpm-software-management/dnf5
+Source0:        %{url}/archive/%{version}/dnf5-%{version}.tar.gz
+# ========== build requires ==========
+BuildRequires:  bash-completion
+BuildRequires:  cmake
+BuildRequires:  doxygen
+BuildRequires:  gettext
+BuildRequires:  pkg-config
+BuildRequires:  toml11-devel
+BuildRequires:  pkgconfig(check)
+BuildRequires:  pkgconfig(fmt)
+BuildRequires:  pkgconfig(json-c)
+BuildRequires:  pkgconfig(libcrypto)
+BuildRequires:  pkgconfig(librepo) >= %{librepo_version}
+BuildRequires:  pkgconfig(libsolv) >= %{libsolv_version}
+BuildRequires:  pkgconfig(libsolvext) >= %{libsolv_version}
+BuildRequires:  pkgconfig(rpm) >= 4.17.0
+BuildRequires:  pkgconfig(sqlite3) >= %{sqlite_version}
+Requires:       libdnf5%{?_isa} = %{version}-%{release}
+Requires:       libdnf5-cli%{?_isa} = %{version}-%{release}
+Recommends:     bash-completion
+%if %{with clang}
```

Comment:
Let's be opinionated on these conditional compile options.  I would rather we set them the way we want them for Mariner than have an overly complex spec file with unsupported features

For example.  we will use gcc-c++ not clang.  I don't think we ever want the server/daemon thing, we probably always want the cli (as examples).  Let's refine and remove the noise and only leave the options we might want to leverage in the future.